pkg/util/errors/errors_test.go

```diff
@@ -75,6 +78,28 @@ func TestIsMatchingSyscallError(t *testing.T) {
 			want: false,
 			err:  &url.Error{Err: io.EOF},
 		},
+		{
+			name:  "azure request error : connection reset",
+			want:  true,
+			match: []syscall.Errno{syscall.ECONNRESET},
+			err: &azure.RequestError{
+				DetailedError: autorest.DetailedError{
+					Original:    os.NewSyscallError("io.read", syscall.ECONNRESET),
+					PackageType: "packageType",
+					Method:      "GET",
+					StatusCode:  "500",
+					Message:     "testing",
+				},
+			},
+		},
+		{
+			name:  "autorest detailed error : connection reset",
+			want:  true,
+			match: []syscall.Errno{syscall.ECONNRESET},
+			err: &autorest.DetailedError{
+				Original: os.NewSyscallError("io.read", syscall.ECONNRESET),
+			},
+		},
```

Comment:
@charlesakalugwu you can see above i am testing both cases. DetailedError and RequestError